Mathlib/Tactic/Widget/Unfold.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2023 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot
+-/
+import Mathlib.Tactic.Widget.SelectPanelUtils
+
+/-! # Unfold widget
+
+This file defines a `unfold?` tactic that displays a widget panel allowing to generate
+an `unfold` call with name specified by selecting a subexpression in the infoview.
+-/
+
+open Lean Meta Server ProofWidgets
+
+/-- If the given expression is a constant, return its name, if it is an application of a constant,
+returns its name, otherwise return `none`. -/
+def Lean.Expr.headName : Expr → Option Name
```

Comment:
I'm also not sure what the good meta style should be here, I think this is the same as doing `e.getAppFn.constName?` (or even `e.getAppFun'.constName?` which is probably a good idea here, it ignores mdata). So, do we need a new def, and if we do should it be defined recursively like this or via the existing ones directly.
I have a vague feeling that reusing existing components would be better in some cases for code generation (smaller dlls, more chance that someone adds a fast `implemented_by` / `extern` and it helps you) but no actual data there.